tensorflow/core/kernels/data/experimental/lmdb_dataset_op_test.cc

```diff
@@ -0,0 +1,322 @@
+        // Attributes
+        {{LMDBDatasetOp::kOutputTypes, dataset_params.output_dtypes},
+         {LMDBDatasetOp::kOutputShapes, dataset_params.output_shapes}});
+    TF_RETURN_IF_ERROR(CreateOpKernel(node_def, op_kernel));
+    return Status::OK();
+  }
+};
+
+// Copy our test data file to the current test program's temporary
+// directory, and return the full path to the copied file.
+// This copying is necessary because LMDB creates lock files adjacent
+// to files that it reads.
+string MaybeCopyDataFile() {
+  string src_loc = io::JoinPath(kDataFileLoc, kDataFileName);
+  string dest_loc = io::JoinPath(testing::TmpDir(), kDataFileName);
+
+  PosixFileSystem fs;
+  // PosixFileSystem::FileExists currently returns Status::OK() if the file
+  // exists and errors::NotFound() if the file doesn't exist. There's no
+  // indication in the code or docs about whether other error codes may be
+  // added in the future, so we code defensively here.
+  Status exists_status = fs.FileExists(dest_loc);
+  if (exists_status.code() == error::NOT_FOUND) {
+    TF_EXPECT_OK(fs.CopyFile(src_loc, dest_loc));
+  } else {
+    TF_EXPECT_OK(exists_status);
+  }
+
+  return dest_loc;
+}
+
+LMDBDatasetParams SingleValidInput() {
+  return {/*filenames=*/{MaybeCopyDataFile()},
+          /*output_dtypes=*/{DT_STRING, DT_STRING},
+          /*output_shapes=*/{PartialTensorShape({}), PartialTensorShape({})},
+          /*node_name=*/kNodeName};
+}
+
+LMDBDatasetParams TwoValidInputs() {
+  return {/*filenames*/ {MaybeCopyDataFile(), MaybeCopyDataFile()},
+          /*output_dtypes*/ {DT_STRING, DT_STRING},
+          /*output_shapes*/ {PartialTensorShape({}), PartialTensorShape({})},
+          /*node_name=*/kNodeName};
+}
+
+LMDBDatasetParams EmptyInput() {
+  return {/*filenames*/ {},
+          /*output_dtypes*/ {DT_STRING, DT_STRING},
+          /*output_shapes*/ {PartialTensorShape({}), PartialTensorShape({})},
+          /*node_name=*/kNodeName};
+}
+
+LMDBDatasetParams InvalidPathAtStart() {
+  return {/*filenames*/ {"This is not a valid filename", MaybeCopyDataFile()},
+          /*output_dtypes*/ {DT_STRING, DT_STRING},
+          /*output_shapes*/ {PartialTensorShape({}), PartialTensorShape({})},
+          /*node_name=*/kNodeName};
+}
+
+LMDBDatasetParams InvalidPathInMiddle() {
+  return {/*filenames*/ {MaybeCopyDataFile(), "This is not a valid filename",
+                         MaybeCopyDataFile()},
+          /*output_dtypes*/ {DT_STRING, DT_STRING},
+          /*output_shapes*/ {PartialTensorShape({}), PartialTensorShape({})},
+          /*node_name=*/kNodeName};
+}
+
+std::vector<Tensor> StringsToTensors(const std::vector<string> strs) {
+  std::vector<Tensor> ret;
+  for (string str : strs) {
+    ret.push_back(CreateTensor<string>(TensorShape({}), {str}));
+  }
+  return ret;
+}
+
+// The tensors we expect to see each time we read through the input
+// data file.
+const std::vector<Tensor> kFileOutput = StringsToTensors({
```

Comment:
The util function [CreateTensors()](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/kernels/data/dataset_test_base.h#L95) can be used here, for example
```C++
const std::vector<Tensor> kFileOutput = CreateTensors<string>(TensorShape({}), {{"0"}, {"a"}, ..., {"j"}};
```

Reply:
Thanks for pointing that out! I have switched to using `CreateTensors` in commit 03140b3.